File: vojaro.web/ClientApp/src/app/api/models/asignatura-filters.ts
import { Carrera } from "./carrera";
import { CarreraOrientacion } from "./carrera-orientacion";
import { Departamento } from "./departamento";
import { Universidad } from "./universidad";

export class AsignaturaFilters {
	id: number | null = null;
	nombre: string | null = null;
	anio: number | null = null;
	cuatrimestre: number | null = null;
	
	universidad: Universidad | null = null;
	departamento: Departamento | null = null;
	carrera: Carrera | null = null;
	carreraOrientacion: CarreraOrientacion | null = null;
}
	

File: vojaro.common/filters/AsignaturaFilters.cs
namespace vojaro.filters
{
    public class AsignaturaFilters
    {
        public long? Id { get; set; }
        public string Nombre { get; set; }
        public int? Anio { get; set; }
        public int? Cuatrimestre { get; set; }
        public long? UniversidadId { get; set; }
        public long? CarreraId { get; set; }
        public long? CarreraOrientacionId { get; set; }
        public TipoFiltroAsignatura TipoFiltro { get; set; } = TipoFiltroAsignatura.Asignatura;


        public enum TipoFiltroAsignatura
        {
            Asignatura = 0,
            Correlativa = 1
        }
    }
}
